pkg/standalone-utils/contracts/relayer/VaultActions.sol

```diff
@@ -160,20 +167,52 @@ abstract contract VaultActions is IBaseRelayerLibrary {
     function _doWeightedExactTokensInForBPTOutReplacements(bytes memory userData) private returns (bytes memory) {
         (uint256[] memory amountsIn, uint256 minBPTAmountOut) = WeightedPoolUserData.exactTokensInForBptOut(userData);
 
-        bool replacedAmounts = false;
+        // Save gas by only re-encoding the data if we actually performed a replacement
+        return
+            _replacedAmounts(amountsIn)
+                ? abi.encode(WeightedPoolUserData.JoinKind.EXACT_TOKENS_IN_FOR_BPT_OUT, amountsIn, minBPTAmountOut)
+                : userData;
+    }
+
+    /**
+     * @dev All versions of stable pool have the same enum and processing for EXACT_TOKENS_IN_FOR_BPT_OUT, so we can
+     * just use the current one. If proportional join is supported, it's a "given out" (parameter is a BPT amount), so
+     * there would be no replacements, and it can be passed through directly to the pool.
+     *
+     * All versions are identical for joins, except the latest ComposableStable appends a proportional join. If you
+     * tried to do a proportional join on any legacy pool, the pool would revert.
```

Comment:
I'd move this comment to the body of the function since it explains how it works, not what it does.

Reply:
Also I found the wording a bit awkward. Maybe try something like:

>The only 'given in' join (in which the parameters are the amounts in) is EXACT_TOKENS_IN_FOR_BPT_OUT, so that is the only one in which we do replacements. Luckily all versions of Stable Pool share the same enum value for it, so we can treat them all at once.
>
>Note that ComposableStablePool v3 supports a proportional join kind which some previous versions did not: while it is not rejected here, if passed to the Pool it will revert.